sphinx_exercise/post_transforms.py

```diff
@@ -0,0 +1,274 @@
+import sphinx.addnodes as sphinx_nodes
+from sphinx.transforms.post_transforms import SphinxPostTransform
+from sphinx.util import logging
+from sphinx.builders.latex import LaTeXBuilder
+from docutils import nodes as docutil_nodes
+
+from .utils import get_node_number, get_refuri, find_parent
+from .nodes import (
+    exercise_enumerable_node,
+    solution_node,
+    exercise_title,
+    exercise_subtitle,
+    solution_title,
+    solution_subtitle,
+)
+
+logger = logging.getLogger(__name__)
+
+
+# Resolver Functions
+
+
+def resolve_enumerated_exercise_node_title(app, node):
+    title = node.children[0]
+    if isinstance(title, exercise_title):
+        node_number = get_node_number(app, node, "exercise")
+        updated_title = exercise_title()
```

Comment:
@AakashGfude perhaps this new `exercise_title` node is detached from the parent?

Reply:
@mmcky it does get attached to the `admonition node` later here: https://github.com/executablebooks/sphinx-exercise/blob/84a3d2ab9eb42600e3f5c1ccef267374ce386580/sphinx_exercise/post_transforms.py#L34

Reply:
it replaces the child for sure -- but that isn't necessarily the same as

```
node += updated_title
```

Reply:
I'll take a quick look

Reply:
@mmcky adding visit, depart exercise_title support for latex here: https://github.com/executablebooks/sphinx-exercise/blob/84a3d2ab9eb42600e3f5c1ccef267374ce386580/sphinx_exercise/__init__.py#L179 , gets rid of the 
``
encountered title node not in section, topic, table, '
                              'admonition or sidebar
``
error, but needs some formatting for proper latex output.

Reply:
Yeah it get's detached as the node metadata isn't retained 😮‍💨 

```
> sphinx-exercise/.tox/py38/lib/python3.8/site-packages/sphinx_exercise/post_transforms.py(25)resolve_enumerated_exercise_node_title()
-> title = node.children[0]
(Pdb) node.children[0]
<exercise_title: <#text: 'Exercise'><exercise_subtitle...>>
(Pdb) node.children[0].parent
<exercise_enumerable_node "exercise-1": <exercise_title...><section...>>
(Pdb) c

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB continue (IO-capturing resumed) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> sphinx-exercise/.tox/py38/lib/python3.8/site-packages/sphinx_exercise/post_transforms.py(39)resolve_enumerated_exercise_node_title()
-> return node
(Pdb) node.children[0].parent
```

Reply:
should be fixed by ebf67f5

Reply:
Thanks a lot @mmcky